server/src/main/java/org/elasticsearch/health/metadata/HealthMetadataService.java

```diff
@@ -82,40 +85,39 @@ private void registerListeners() {
         }
 
         ClusterSettings clusterSettings = clusterService.getClusterSettings();
-        clusterSettings.addSettingsUpdateConsumer(
+        Stream.of(
             CLUSTER_ROUTING_ALLOCATION_HIGH_DISK_WATERMARK_SETTING,
-            value -> updateOnSettingsUpdated(CLUSTER_ROUTING_ALLOCATION_HIGH_DISK_WATERMARK_SETTING.getKey(), value.getStringRep())
-        );
-        clusterSettings.addSettingsUpdateConsumer(
             CLUSTER_ROUTING_ALLOCATION_DISK_FLOOD_STAGE_WATERMARK_SETTING,
-            value -> updateOnSettingsUpdated(CLUSTER_ROUTING_ALLOCATION_DISK_FLOOD_STAGE_WATERMARK_SETTING.getKey(), value.getStringRep())
-        );
-        clusterSettings.addSettingsUpdateConsumer(
-            CLUSTER_ROUTING_ALLOCATION_DISK_FLOOD_STAGE_FROZEN_WATERMARK_SETTING,
-            value -> updateOnSettingsUpdated(
-                CLUSTER_ROUTING_ALLOCATION_DISK_FLOOD_STAGE_FROZEN_WATERMARK_SETTING.getKey(),
-                value.getStringRep()
-            )
-        );
-        clusterSettings.addSettingsUpdateConsumer(
+            CLUSTER_ROUTING_ALLOCATION_DISK_FLOOD_STAGE_FROZEN_WATERMARK_SETTING
+        )
+            .forEach(
+                setting -> clusterSettings.addSettingsUpdateConsumer(
+                    setting,
+                    value -> updateOnSettingsUpdated(new UpdateDiskMetadata(setting.getKey(), value.getStringRep()))
+                )
+            );
```

Comment:
Is this the same as the below `forEach` ? 

If they are can we just have one `Stream.of` for all disk related values?

Reply:
Sadly no, they are handling `ByteSizeValue` and `RelativeByteSizeValue`, which aren't sharing the same ancestor, so it's not possible to abstract into 1 stream :(

Reply:
Makes sense, thanks :) (my eyes glanced over that difference)